### DISCUSSION TITLE
Is there a way to test that the down script works?

### DISCUSSION THREAD

**Discussion by badaz** [haugene/docker-transmission-openvpn#2833] [Q&A]
I've checked the docs but could not find an answer to that question: is it possible to test that the down script works as expected (the one that is injected in the provided openvpn configuration and that is supposed to stop transmission when the VPN tunnel is down).

I tried to "docker exec" inside the container, then kill the openvpn process by calling kill ${OPENVPN_PID}, which resulted in calling the down script that stopped transmission, then the container stopped as a result of having no running process anymore (and restarted since I have set restart policy "always"). Is this the proper way to test this ? Or is there a way to fake the tunnel down condition instead of completely killing the openvpn process?

Bonus question : is openvpn killed when the connection to the VPN is lost? If not then what is the state of openvpn in the container after having lost its connection? Will it stay disconnected until one restarts the container? Will it try to reconnect?

**Comment by edgd1er** [comment]
I guess the way to test is to ask openvpn to disconnect and watch for the route down event and the down script being executed.

The image at the moment doesn t have the management interface enabled. You could perform the modification on configuration to enable that interface and perfoms the test.

https://openvpn.net/community-resources/management-interface/

**Reply by badaz** [reply]
Thanks, I didn't know about that interface. I'll try that when I find the time.
